apps/src/codebridge/FileBrowser/index.tsx

```diff
@@ -762,13 +743,45 @@ export const FileBrowser = React.memo(() => {
     () => (e: DragOverEvent) => {
       if (e?.over) {
         if (e.active.data.current?.type === DragType.FOLDER) {
-          moveFolder(e.active.data.current.id as string, e.over.id as string);
+          const duplicate = checkForDuplicateFoldername(
+            project.folders[e.active.data.current.id].name,
+            e.over.id as string,
+            project.folders
+          );
+          if (duplicate) {
+            dialogControl?.showDialog({
+              type: DialogType.GenericAlert,
+              title: duplicate,
+            });
+          } else {
+            moveFolder(e.active.data.current.id as string, e.over.id as string);
+          }
         } else if (e.active.data.current?.type === DragType.FILE) {
-          moveFile(e.active.data.current.id as string, e.over.id as string);
+          const duplicate = checkForDuplicateFilename(
+            project.files[e.active.data.current.id].name,
+            e.over.id as string,
+            project.files
+          );
+          if (duplicate) {
+            dialogControl?.showDialog({
+              type: DialogType.GenericAlert,
+              title: duplicate,
+            });
+          } else {
+            moveFile(e.active.data.current.id as string, e.over.id as string);
+          }
```

Comment:
these duplicate checks should've gone into the prior DnD PR. I don't know what happened to that commit that included them, which is also the one that had the `DragType` enum originally.